pkg/util/schedulerlatency/sampler.go

```diff
@@ -332,15 +333,10 @@ func percentile(h *metrics.Float64Histogram, p float64) float64 {
 		}
 	}
 
-	return (start + end) / 2 // grab the mid-point within the bucket
-
-	// TODO(irfansharif): Implement proper linear interpolation instead and then
-	// write test comparing against it against metric.ValueAtQuantileWindowed.
-	// If pmax, we'll return the bucket max. If pmin, we'll return the bucket
-	// min. If the 90th and 100th percentile values lie within some bucket, and
-	// we're looking for 99.9th percentile, we'll interpolate to the 99% point
-	// between bucket start and end. Because we're doing this naive mid-point
-	// thing, it makes for a confusing difference when comparing the p99
-	// computed off of go.scheduler_latency vs.
-	// admission.scheduler_latency_listener.p99_nanos.
+	// Find the target rank within bucket boundaries.
+	// NB: The 0.5 is added for rounding purposes; it helps in cases where
```

Comment:
@irfansharif After doing this here, I am wondering whether my decision in util/metric.go was correct and if this is the strategy we want. By rounding it to the nearest int, I think we are assuming the data within the bucket is equally spaced. For example, if we have 5 samples in a bucket with a lower bound of 0 and an upper bound of 10. We assume the data is [0, 2.5, 5, 7.5, 10]. Here, 99th and 100th percentile would return 10 instead of 9.9 for the 99th. I initially thought this was ok, but not so sure anymore. It probably doesn't matter when buckets have large counts as this smooths itself out, but something to consider and we should keep both parts consistent with each other.

FWIW prometheus client does not do this: https://github.com/prometheus/prometheus/blob/d9162189/promql/quantile.go#L100